<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="John Barlow" />

<meta name="date" content="2018-04-23" />

<title>made4 Multivariate analysis of microarray data</title>

<script src="made4_demo_script_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="made4_demo_script_files/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="made4_demo_script_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="made4_demo_script_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="made4_demo_script_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="made4_demo_script_files/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="fluid-row" id="header">



<h1 class="title toc-ignore">made4 Multivariate analysis of microarray data</h1>
<h4 class="author"><em>John Barlow</em></h4>
<h4 class="date"><em>April 23, 2018</em></h4>

</div>


<div id="references" class="section level2">
<h2>references:</h2>
<p>I used the following resources to learn about using made4</p>
<ol style="list-style-type: decimal">
<li>Bioconductor made4</li>
</ol>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/made4.html" class="uri">https://www.bioconductor.org/packages/release/bioc/html/made4.html</a></p>
<p><a href="http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf" class="uri">http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf</a></p>
<ol start="2" style="list-style-type: decimal">
<li>Culhane AC, Thioulouse J, Perriere G, Higgins DG.(2005) MADE4: an R package for multivariate analysis of gene expression data. Bioinformatics 21(11):2789-90. note the introduction and reference manual can be opened from within R</li>
</ol>
<p>“made4 is useful for Multivariate data analysis and graphical display of microarray data. Functions include between group analysis and coinertia analysis. It contains functions that require ade4 package.”</p>
<p>Korin found this package because we are particularly interested in the coinertia analysis function to explore potential trends between gene abundancy data sets.</p>
</div>
<div id="before-the-preliminaries" class="section level2">
<h2>before the preliminaries</h2>
<p>recent versions of made4 and associated packages were built under R version 3.4.4, so like me you might need to upgrade from an older R version or else when you download made4 it will throw a warning message - I don’t know if updating was absolutely necessary, but I did not chance it as my last R update was about 6 months old</p>
<p>I found a package to make the process of upgrading R a bit easier - the installr package - it seemed to work well and included an option to upgrade packages when it ran</p>
<p>you need to run installr in the R Gui console, not from R studio here is a link on updating R with installr</p>
<p><a href="http://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html">ttp://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html</a></p>
<p>you can install the installr package from RStudio</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;installr&quot;</span>)</code></pre></div>
<p>then close RStudio and open RGui in RGui open the installr library and run updateR</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(installr)
<span class="kw">updateR</span>()</code></pre></div>
</div>
<div id="the-preliminaries" class="section level2">
<h2>the preliminaries</h2>
<p>download made4 package - made4 package uses ade4 package, scatterplot3d package, RColorBrewer and gplots</p>
<p>once your R version is up to date, then on to the preliiminaries of installing made4 and associated packages - ade4 and scatterplot3d appear to install automatically when you install made4</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;made4&quot;</span>)</code></pre></div>
<p>Korin suggests one needs to install made4 as below from the biocLite not via install packages, but install packages seemed to work for me (although it loaded an older version and to get the most recent version I had to download direct from the made4 bioconductor site</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;https://bioconductor.org/biocLite.R&quot;</span>)
<span class="kw">biocLite</span>(<span class="st">&quot;made4&quot;</span>)
<span class="kw">biocLite</span>(<span class="st">&quot;BiocUpgrade&quot;</span>)</code></pre></div>
<p>confirm that the 3 downloaded files are in your current working directory</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getwd</span>()
<span class="co">#move files into current wd or if needed set wd</span>
<span class="kw">setwd</span>(<span class="st">&quot;insert address for your particular wd&quot;</span>)</code></pre></div>
<p>load made4 - associated packages ade4, RColorBrewer, gplots, and scatterplot3d load automatically when you load made4</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(made4)</code></pre></div>
<div id="data-structure-constraints-for-made4" class="section level4">
<h4>Data structure constraints for made4:</h4>
<ol style="list-style-type: decimal">
<li>with some functions for testing relationships between 2 data sets is that the datasets must contain the same samples (“either the rows or the columns of a matrix must be matchable”).</li>
</ol>
<p>read in the 3 data files</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_mat &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;res_mat_abun.csv&quot;</span>)
bac_mat &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;bac_mat_abun.csv&quot;</span>)
vf_mat &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;vf_mat_abun.csv&quot;</span>)</code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>made4 needs the abundance values to be numeric, and for some reason a number of values are as integers in our csv files - so code we figured out to correct this</li>
</ol>
<p>check the structure of the data files</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(res_mat)
<span class="co">#so clean thus up by setting all values to numeric, except not the gene names</span>
res_mat_all &lt;-<span class="st"> </span>res_mat[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>)]
res_mat_all[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(res_mat_all[], as.numeric)
<span class="kw">rownames</span>(res_mat_all) &lt;-<span class="st"> </span>res_mat<span class="op">$</span>Name
<span class="kw">head</span>(res_mat_all)
<span class="kw">str</span>(res_mat_all)</code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>also, a number of functions throw an error if any sites have all 0 values for the abundance of observed genes e.g. the HOSP, WOCA, and SWCA for ARGs all have zero abundance for ARGs (see <a href="fig3_arg_freq_by_site.html">Figure 3 ARG freq by site.html</a>).</li>
</ol>
<p>here is the code Korin figured out to resolve this issue</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">none &lt;-<span class="st"> </span><span class="kw">lapply</span>(res_mat_all, <span class="cf">function</span>(x) <span class="kw">all</span>(x <span class="op">==</span><span class="st"> </span><span class="dv">0</span>))
<span class="kw">which</span>(none <span class="op">==</span><span class="st"> &quot;TRUE&quot;</span>)
<span class="co"># remove those 6 columns</span>
res_mat2 &lt;-<span class="st"> </span>res_mat_all[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">15</span>,<span class="dv">21</span>,<span class="dv">24</span>)]
<span class="kw">head</span>(res_mat_all)
<span class="kw">str</span>(res_mat_all)</code></pre></div>
<p>repeat for the other 2 files, removing the same 6 columns from each file</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bac_mat_all &lt;-<span class="st"> </span>bac_mat[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>)]
bac_mat_all[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(bac_mat_all[], as.numeric)
<span class="kw">rownames</span>(bac_mat_all) &lt;-<span class="st"> </span>bac_mat<span class="op">$</span>Name
bac_mat2 &lt;-<span class="st"> </span>bac_mat_all[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">15</span>,<span class="dv">21</span>,<span class="dv">24</span>)]
<span class="kw">head</span>(bac_mat2)
<span class="kw">str</span>(bac_mat2)

vf_mat_all &lt;-<span class="st"> </span>vf_mat[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>)]
vf_mat_all[] &lt;-<span class="st"> </span><span class="kw">lapply</span>(vf_mat_all[], as.numeric)
<span class="kw">rownames</span>(vf_mat_all) &lt;-<span class="st"> </span>vf_mat<span class="op">$</span>Name
vf_mat2 &lt;-<span class="st"> </span>vf_mat_all[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">15</span>,<span class="dv">21</span>,<span class="dv">24</span>)]
<span class="kw">head</span>(vf_mat2)
<span class="kw">str</span>(vf_mat2)</code></pre></div>
</div>
<div id="data-files-now-ready-to-use" class="section level4">
<h4>data files now ready to use</h4>
</div>
</div>
<div id="made4-functions" class="section level2">
<h2><em>made4</em> functions</h2>
<p>made4 is well documented - to view documentation</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>(<span class="st">&quot;made4&quot;</span>)</code></pre></div>
<p>the package also contains a data set of gene expression data</p>
<p>made4 has an overview function which generates a boxplot, histogram and hierachial tree of the data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">overview</span>(res_mat2)</code></pre></div>
<p>the ord function makes it east to run ordination methods to explore structure of a data matrix - the methods include correspondence (coa, the default), non-symmetric correspondence analysis (nsc) or principal component (pca).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_coa&lt;-<span class="kw">ord</span>(res_mat2, <span class="dt">type=</span> <span class="st">&quot;coa&quot;</span>)

<span class="kw">summary</span>(res_coa<span class="op">$</span>ord)</code></pre></div>
<p>and plot the results</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(res_coa)

<span class="kw">plotgenes</span>(res_coa, <span class="dt">nlab=</span><span class="dv">1</span>) <span class="co">#if you include &quot;nlab&quot; </span>
<span class="kw">plotarrays</span>(res_coa, <span class="dt">graph=</span><span class="st">&quot;groups&quot;</span>) <span class="co">#&quot;simple&quot; gives the points, &quot;groups&quot; adds the labels</span>

<span class="co">#you can generate a list of the variables (genes) at the end of the axis or gradient</span>
<span class="kw">topgenes</span>(res_coa, <span class="dt">axis=</span><span class="dv">1</span>, <span class="dt">n=</span><span class="dv">5</span>, <span class="dt">end=</span><span class="st">&quot;pos&quot;</span>)

<span class="co"># and compare lists of top genes at the ends of gradients</span>
a&lt;-<span class="kw">topgenes</span>(res_coa, <span class="dt">axis=</span><span class="dv">2</span>, <span class="dt">n=</span><span class="dv">10</span>, <span class="dt">end=</span><span class="st">&quot;pos&quot;</span>)
b&lt;-<span class="kw">topgenes</span>(res_coa, <span class="dt">axis=</span><span class="dv">3</span>, <span class="dt">n=</span><span class="dv">10</span>, <span class="dt">end=</span><span class="st">&quot;pos&quot;</span>)
z&lt;-<span class="kw">comparelists</span>(a,b)
<span class="kw">print.comparelists</span>(z)
z<span class="op">$</span>intersect
z<span class="op">$</span>Set.Diff


<span class="co">#including generating in 3d</span>
<span class="kw">do3d</span>(res_coa<span class="op">$</span>ord<span class="op">$</span>li)
<span class="kw">do3d</span>(res_coa<span class="op">$</span>ord<span class="op">$</span>co)</code></pre></div>
<p>the support documents provide package and demo code for visualizing the 3d plots dynamically (can be rotated).</p>
<div id="coinertia-analysis" class="section level4">
<h4>Coinertia Analysis</h4>
<p>exploring trends or co-relationships between multiple datasets</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#bacteria and ARGs</span>
c &lt;-<span class="st"> </span><span class="kw">cia</span>(bac_mat2, res_mat2)
c<span class="op">$</span>coinertia<span class="op">$</span>RV
<span class="co">#0.445</span>
<span class="kw">plot.cia</span>(c)

<span class="co"># virulence and ARGs</span>
c2 &lt;-<span class="st"> </span><span class="kw">cia</span>(vf_mat2, res_mat2)
c2<span class="op">$</span>coinertia<span class="op">$</span>RV
<span class="co">#0.647</span>
<span class="kw">plot.cia</span>(c2)


<span class="co"># virulence and bacteria</span>
c3 &lt;-<span class="st"> </span><span class="kw">cia</span>(vf_mat2, bac_mat2)
c3<span class="op">$</span>coinertia<span class="op">$</span>RV
<span class="co">#0.358</span>
<span class="kw">plot.cia</span>(c3)</code></pre></div>
<p>The RV coefficient is a measure of similarity between the 2 datasest and is number between 0 and 1, with closer to 1 being more similarity betwen the structure of the data sets.</p>
<p>On the plot, the point is the gene abundance score for a case from one data set and the arrow is the gene abundance score form the other data set. The lines join these 2 pints and repesent the amunt of divergence between gene abundance profiles for the 2 data sets. A short line = good agreement within a site when comparing the structure of the abundance data from the 2 data sets.</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
