<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="John Barlow" />

<meta name="date" content="2018-04-23" />

<title>made4 Multivariate analysis of microarray data</title>

<script src="made4_demo_script_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="made4_demo_script_files/bootstrap-3.3.6/css/bootstrap.min.css" rel="stylesheet" />
<script src="made4_demo_script_files/bootstrap-3.3.6/js/bootstrap.min.js"></script>
<script src="made4_demo_script_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="made4_demo_script_files/navigation-1.1/tabsets.js"></script>
<script src="made4_demo_script_files/navigation-1.1/codefolding.js"></script>
<link href="made4_demo_script_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="made4_demo_script_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="made4_demo_script_files/clean-0.1/clean.css" rel="stylesheet" />
<script src="made4_demo_script_files/clean-0.1/clean.js"></script>




<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
</style>
<div class="container-fluid main-container">
  <div class="row">
     <div class="col-md-10">




<div id="header">
<h1 class="title">made4 Multivariate analysis of microarray data</h1>
<h4 class="author"><em>John Barlow</em></h4>
<h4 class="date"><em>April 23, 2018</em></h4>
</div>

<div id="content">
<div id="references" class="section level1">
<h1>REFERENCES:</h1>
</div>
<div id="i-used-the-following-resources-to-learn-about-using-made4" class="section level1">
<h1>I used the following resources to learn about using made4</h1>
</div>
<div id="bioconductor-made4" class="section level1">
<h1>1. Bioconductor made4</h1>
</div>
<div id="httpswww.bioconductor.orgpackagesreleasebiochtmlmade4.html" class="section level1">
<h1><a href="https://www.bioconductor.org/packages/release/bioc/html/made4.html" class="uri">https://www.bioconductor.org/packages/release/bioc/html/made4.html</a></h1>
</div>
<div id="httpwww.bioconductor.orgpackages2.7biocvignettesmade4instdocintroduction.pdf" class="section level1">
<h1><a href="http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf" class="uri">http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf</a></h1>
</div>
<div id="culhane-ac-thioulouse-j-perriere-g-higgins-dg.2005-made4-an-r-package-for-multivariate-analysis-of-gene-expression-data.-bioinformatics-21112789-90." class="section level1">
<h1>2. Culhane AC, Thioulouse J, Perriere G, Higgins DG.(2005) MADE4: an R package for multivariate analysis of gene expression data. Bioinformatics 21(11):2789-90.</h1>
</div>
<div id="note-the-introduction-and-reference-manual-can-be-opened-from-within-r" class="section level1">
<h1>note the introduction and reference manual can be opened from within R</h1>
</div>
<div id="section" class="section level1">
<h1></h1>
</div>
<div id="made4-is-useful-for-multivariate-data-analysis-and-graphical-display-of-microarray-data.-functions-include-between-group-analysis-and-coinertia-analysis.-it-contains-functions-that-require-ade4-package." class="section level1">
<h1>“made4 is useful for Multivariate data analysis and graphical display of microarray data. Functions include between group analysis and coinertia analysis. It contains functions that require ade4 package.”</h1>
</div>
<div id="section-1" class="section level1">
<h1>———————————————————————</h1>
</div>
<div id="before-the-preliminaries" class="section level1">
<h1>before the preliminaries</h1>
</div>
<div id="recent-versions-of-made4-and-associated-packages-were-built-under-r-version-3.4.4-so-like-me-you-might-need-to-upgrade-from-an-older-r-version-or-else-when-you-download-made4-it-will-throw-a-warning-message---i-dont-know-if-updating-was-absolutely-necessary-but-i-did-not-chance-it-as-my-last-r-update-was-about-6-months-old" class="section level1">
<h1>recent versions of made4 and associated packages were built under R version 3.4.4, so like me you might need to upgrade from an older R version or else when you download made4 it will throw a warning message - I don’t know if updating was absolutely necessary, but I did not chance it as my last R update was about 6 months old…</h1>
</div>
<div id="i-found-a-package-to-make-the-process-of-upgrading-r-a-bit-easier---the-installr-package---it-seemed-to-work-well-and-included-an-option-to-upgrade-packages-when-it-ran" class="section level1">
<h1>I found a package to make the process of upgrading R a bit easier - the installr package - it seemed to work well and included an option to upgrade packages when it ran</h1>
</div>
<div id="preference-is-to-run-installr-in-the-r-gui-console-not-from-r-studio" class="section level1">
<h1>preference is to run installr in the R Gui console, not from R studio</h1>
</div>
<div id="here-is-a-link-on-updating-r-with-installr-httpbioinfo.umassmed.edubootstrappersbootstrappers-coursescoursesrcourseadditional_resourcesupdating_r.html" class="section level1">
<h1>here is a link on updating R with installr <a href="http://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html" class="uri">http://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html</a></h1>
</div>
<div id="you-can-install-the-installr-package-from-rstudio" class="section level1">
<h1>you can install the installr package from RStudio</h1>
<p>install.packages(“installr”) # then close RStudio and open RGui # in RGui open the installr library and run updateR # library(installr) # updateR()</p>
</div>
<div id="section-2" class="section level1">
<h1>———————————————————————-</h1>
</div>
<div id="the-preliminaries" class="section level1">
<h1>the preliminaries</h1>
</div>
<div id="download-made4-package---made4-package-uses-ade4-package-scatterplot3d-package-rcolorbrewer-and-gplots" class="section level1">
<h1>download made4 package - made4 package uses ade4 package, scatterplot3d package, RColorBrewer and gplots</h1>
</div>
<div id="once-your-r-version-is-up-to-date-then-on-to-the-preliiminaries-of-installing-made4-and-associated-packages---ade4-and-scatterplot3d-appear-to-install-automatically-when-you-install-made4" class="section level1">
<h1>once your R version is up to date, then on to the preliiminaries of installing made4 and associated packages - ade4 and scatterplot3d appear to install automatically when you install made4</h1>
<p>install.packages(“made4”) # ggtree and ape are a couple phylogenetics packages I have been exploring for our work, # not using in this exercise for now #install.packages(“ggtree”) # ggtree not available for most recent version # install.packages(“ape”)</p>
<div id="korin-suggests-one-needs-to-install-made4-as-below-from-the-bioclite-source-not-via-install.packages-but-install.packages-seemed-to-work-for-me-although-it-loaded-an-older-version-and-to-get-the-most-recent-version-i-had-to-download-direct-from-the-made4-bioconductor-site" class="section level2">
<h2>Korin suggests one needs to install made4 as below from the biocLite source, not via install.packages(), but install.packages seemed to work for me (although it loaded an older version and to get the most recent version I had to download direct from the made4 bioconductor site</h2>
</div>
</div>
<div id="sourcehttpsbioconductor.orgbioclite.r" class="section level1">
<h1>source(“<a href="https://bioconductor.org/biocLite.R" class="uri">https://bioconductor.org/biocLite.R</a>”)</h1>
</div>
<div id="bioclitemade4" class="section level1">
<h1>biocLite(“made4”)</h1>
</div>
<div id="bioclitebiocupgrade" class="section level1">
<h1>biocLite(“BiocUpgrade”)</h1>
</div>
</div>


    </div>
    <div class="col-md-2">
      <div id="toc">
      	<button type="button" class="close">×</button>
        <ul>
        <li><a href="#references">REFERENCES:</a></li>
        <li><a href="#i-used-the-following-resources-to-learn-about-using-made4">I used the following resources to learn about using made4</a></li>
        <li><a href="#bioconductor-made4">1. Bioconductor made4</a></li>
        <li><a href="#httpswww.bioconductor.orgpackagesreleasebiochtmlmade4.html"><a href="https://www.bioconductor.org/packages/release/bioc/html/made4.html" class="uri">https://www.bioconductor.org/packages/release/bioc/html/made4.html</a></a></li>
        <li><a href="#httpwww.bioconductor.orgpackages2.7biocvignettesmade4instdocintroduction.pdf"><a href="http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf" class="uri">http://www.bioconductor.org/packages//2.7/bioc/vignettes/made4/inst/doc/introduction.pdf</a></a></li>
        <li><a href="#culhane-ac-thioulouse-j-perriere-g-higgins-dg.2005-made4-an-r-package-for-multivariate-analysis-of-gene-expression-data.-bioinformatics-21112789-90.">2. Culhane AC, Thioulouse J, Perriere G, Higgins DG.(2005) MADE4: an R package for multivariate analysis of gene expression data. Bioinformatics 21(11):2789-90.</a></li>
        <li><a href="#note-the-introduction-and-reference-manual-can-be-opened-from-within-r">note the introduction and reference manual can be opened from within R</a></li>
        <li><a href="#section"></a></li>
        <li><a href="#made4-is-useful-for-multivariate-data-analysis-and-graphical-display-of-microarray-data.-functions-include-between-group-analysis-and-coinertia-analysis.-it-contains-functions-that-require-ade4-package.">“made4 is useful for Multivariate data analysis and graphical display of microarray data. Functions include between group analysis and coinertia analysis. It contains functions that require ade4 package.”</a></li>
        <li><a href="#section-1">———————————————————————</a></li>
        <li><a href="#before-the-preliminaries">before the preliminaries</a></li>
        <li><a href="#recent-versions-of-made4-and-associated-packages-were-built-under-r-version-3.4.4-so-like-me-you-might-need-to-upgrade-from-an-older-r-version-or-else-when-you-download-made4-it-will-throw-a-warning-message---i-dont-know-if-updating-was-absolutely-necessary-but-i-did-not-chance-it-as-my-last-r-update-was-about-6-months-old">recent versions of made4 and associated packages were built under R version 3.4.4, so like me you might need to upgrade from an older R version or else when you download made4 it will throw a warning message - I don’t know if updating was absolutely necessary, but I did not chance it as my last R update was about 6 months old…</a></li>
        <li><a href="#i-found-a-package-to-make-the-process-of-upgrading-r-a-bit-easier---the-installr-package---it-seemed-to-work-well-and-included-an-option-to-upgrade-packages-when-it-ran">I found a package to make the process of upgrading R a bit easier - the installr package - it seemed to work well and included an option to upgrade packages when it ran</a></li>
        <li><a href="#preference-is-to-run-installr-in-the-r-gui-console-not-from-r-studio">preference is to run installr in the R Gui console, not from R studio</a></li>
        <li><a href="#here-is-a-link-on-updating-r-with-installr-httpbioinfo.umassmed.edubootstrappersbootstrappers-coursescoursesrcourseadditional_resourcesupdating_r.html">here is a link on updating R with installr <a href="http://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html" class="uri">http://bioinfo.umassmed.edu/bootstrappers/bootstrappers-courses/courses/rCourse/Additional_Resources/Updating_R.html</a></a></li>
        <li><a href="#you-can-install-the-installr-package-from-rstudio">you can install the installr package from RStudio</a></li>
        <li><a href="#section-2">———————————————————————-</a></li>
        <li><a href="#the-preliminaries">the preliminaries</a></li>
        <li><a href="#download-made4-package---made4-package-uses-ade4-package-scatterplot3d-package-rcolorbrewer-and-gplots">download made4 package - made4 package uses ade4 package, scatterplot3d package, RColorBrewer and gplots</a></li>
        <li><a href="#once-your-r-version-is-up-to-date-then-on-to-the-preliiminaries-of-installing-made4-and-associated-packages---ade4-and-scatterplot3d-appear-to-install-automatically-when-you-install-made4">once your R version is up to date, then on to the preliiminaries of installing made4 and associated packages - ade4 and scatterplot3d appear to install automatically when you install made4</a><ul>
        <li><a href="#korin-suggests-one-needs-to-install-made4-as-below-from-the-bioclite-source-not-via-install.packages-but-install.packages-seemed-to-work-for-me-although-it-loaded-an-older-version-and-to-get-the-most-recent-version-i-had-to-download-direct-from-the-made4-bioconductor-site">Korin suggests one needs to install made4 as below from the biocLite source, not via install.packages(), but install.packages seemed to work for me (although it loaded an older version and to get the most recent version I had to download direct from the made4 bioconductor site</a></li>
        </ul></li>
        <li><a href="#sourcehttpsbioconductor.orgbioclite.r">source(“<a href="https://bioconductor.org/biocLite.R" class="uri">https://bioconductor.org/biocLite.R</a>”)</a></li>
        <li><a href="#bioclitemade4">biocLite(“made4”)</a></li>
        <li><a href="#bioclitebiocupgrade">biocLite(“BiocUpgrade”)</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

<script>
$(document).ready(function () {

  // add bootstrap table styles to pandoc tables
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');

});
</script>

<script>
$(document).ready(function () {
 	    $('#content img')
 	  .addClass("image-lb");
  $('#content').magnificPopup({
	      type:'image',
	      closeOnContentClick: false,
	      delegate: 'img',
	      gallery: {enabled: false },
	      image: {
	        verticalFit: true,
          titleSrc: 'alt'
	      }
 	    });
 	});
</script>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
